iceoryx_utils/source/posix_wrapper/access_control.cpp

```diff
@@ -79,20 +79,12 @@ AccessController::smartAclPointer_t AccessController::createACL(const int32_t f_
     auto aclInitCall = cxx::makeSmartC(
         acl_init, cxx::ReturnMode::PRE_DEFINED_ERROR_CODE, {static_cast<acl_t>(nullptr)}, {}, f_numEntries);
 
-    if (aclInitCall.hasErrors())
-    {
-        std::cerr << "Error: Could not allocate new ACL." << std::endl;
-        std::terminate();
-    }
+    cxx::Ensures(!aclInitCall.hasErrors() && " Could not allocate new ACL");
```

Comment:
Here it makes more sense to return a `cxx::expected` and let the user handle the error.
Everytime we hide some kind of termination inside an `cxx::Ensures`, `cxx::Expects` we increase the likelyhood that some unexpected termination happens due to a recoverable error.

But if we return an expected the compiler enforces error handling to the user and the user has to make sure that all requirements are met.
So we do not hide the error, we let the user face it.

Reply:
I agree with you in this case! Adapted the code accordingly.